src/coreclr/src/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/FieldAccessors/StaticFieldAccessor.cs

```diff
@@ -51,6 +56,11 @@ public sealed override void SetFieldDirect(TypedReference typedReference, object
             {
                 RuntimeAugments.EnsureClassConstructorRun(_cctorContext);
             }
+            if (IsFieldInitOnly)
+            {
+                Debug.Fail("Cannot set field with InitOnly flag set.");
+                throw new InvalidOperationException();
```

Comment:
Could you make this a FieldAccessException? It's what CoreCLR throws. Also, no need for the Debug.Fail - this is not breaking internal invariants - it's simply invalid use of the API that we handle.

Let's put "Cannot set initonly static field after its owning type is initialized.'" as the message.

Could you also put the message into a resource string? LiteralFieldAccessor.cs has a sample - you basically need to add it to Strings.resx of the project.